vpr/SRC/route/route_tree_timing.c

```diff
@@ -451,12 +465,81 @@ static void load_rt_subtree_Tdel(t_rt_node * subtree_rt_root, float Tarrival) {
 
 		Tchild = Tdel + g_rr_switch_inf[iswitch].R * child_node->C_downstream;
 		Tchild += g_rr_switch_inf[iswitch].Tdel; /* Intrinsic switch delay. */
-		load_rt_subtree_Tdel(child_node, Tchild);
+		load_route_tree_Tdel(child_node, Tchild);
 
 		linked_rt_edge = linked_rt_edge->next;
 	}
 }
 
+static void load_route_tree_rr_route_inf(const t_rt_node* root, int prev_node, int prev_edge) {
+
+	/* Traverses down a route tree and updates rr_node_inf for all nodes *
+	 * to reflect that these nodes have already been routed to 			 */
+
+	INCREMENTAL_REROUTING_ASSERT(root != nullptr);
+
+	t_linked_rt_edge* edge {root->u.child_list};
+	
+	for (;;) {
+		int inode = root->inode;
+		rr_node_route_inf[inode].prev_node = NO_PREVIOUS;
+		rr_node_route_inf[inode].prev_edge = NO_PREVIOUS;
+		// path cost should be HUGE_POSITIVE_FLOAT
+		INCREMENTAL_REROUTING_ASSERT(equal_approx(rr_node_route_inf[inode].path_cost, HUGE_POSITIVE_FLOAT));
+
+		// reached a sink
+		if (!edge) {return;}
+		// join point (sibling edges)
+		else if (edge->next) {
+			int edge_num = 0;
+			// recursively update for each of its sibling branches
+			do {
+				load_route_tree_rr_route_inf(edge->child, inode, edge_num);
+				++edge_num;
+				edge = edge->next;
+			} while (edge);
+			return;
+		}
+
+		root = edge->child;
+		edge = root->u.child_list;
+		prev_node = inode;
+	}	
+}
+void load_route_tree_rr_route_inf(t_rt_node* root) {	// from root
```

Comment:
Using an overloaded function here is confusing.  Could this be done in one function?